File: Codes/Hash/Non-Cryptographic/BPHash/BPHash.vb
'  Hash function by Bruno Preiss
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function BPHash (key As String) As UInteger
    Dim state As UInteger = 0
	Dim i As UInteger = 0

	For i = 0 To key.Length - 1
		state = state << 7 Xor CByte(AscW(key(CInt(i))))
	Next

	Return state
End Function

File: Codes/Hash/Non-Cryptographic/BPHash/BPHash.java
/*
	Hash function by Bruno Preiss
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static long BPHash(String key)
{
    long state = 0;

    for(int i = 0; i < key.length(); i++)
    {
        state = state << 7 ^ key.charAt(i);
    }

    return state;
}